### PR TITLE
bearlib: Add deprecate_bear decorator

### PR DESCRIPTION
**Pull request by sils**

### CODE REVIEW

File: coalib/bearlib/__init__.py [hunk 1]
```diff
@@ -97,3 +99,41 @@ def wrapping_function(*args, **kwargs):
...
+
+def deprecate_bear(bear):
+    """
+    Use this to deprecate a bear. Say we have a bear:
+
+    >>> class SomeBear:
+    ...     def run(*args):
+    ...         print("I'm running!")
+
+    Now, say you have still users that use ``SomeOldBear`` which is doing the
```
Comment (Adrianzatreanu) [inline coalib/bearlib/__init__.py]: badly formulated, try
"Supposing someone still tries to run SomeOldBear, you can decorate it using.."

File: coalib/bearlib/__init__.py [hunk 2]
```diff
@@ -97,3 +99,41 @@ def wrapping_function(*args, **kwargs):
...
+
+    >>> import sys
+    >>> logging.basicConfig(stream=sys.stdout, level=logging.DEBUG)
+    >>> SomeOldBear().run()
+    WARNING:root:The bear SomeOldBear is deprecated. Use SomeBear instead!
+    I'm running!
+
+    :param bear: An old bear class that inherits from the new one (so it gets
+                 its methods and can just contain a pass.)
+    :return: A bear class that warns about deprecation on use.
```
Comment (Asnelchristian) [inline coalib/bearlib/__init__.py]: align the :return: docs with the one of the setting above 
anyway this can be a newcomer issue :sweat_smile:

**Comment by Asnelchristian** [comment]
ack https://github.com/coala/coala/pull/2989/commits/80c815df07786a3d64e521d31360701490024eac 
https://github.com/coala/coala/pull/2989/commits/7a375abc60ebd1bf54bb1378cf0e410d3bffd255

File: coalib/bearlib/__init__.py (hunk 2 above)
Comment (sils) [inline coalib/bearlib/__init__.py]: cool, file one

**Comment by sils** [comment]
reack 8145c13 deprecate_settings: Use builtin logging
bdde967 bearlib: Add deprecate_bear decorator

**Comment by Asnelchristian** [comment]
needs a rebase

**Comment by sils** [comment]
just did that

**Comment by sils** [comment]
@rultor merge

**Pull request merged**